python/ray/serve/_private/version.py

```diff
@@ -37,7 +65,44 @@ def __eq__(self, other: Any) -> bool:
 
     def to_proto(self) -> bytes:
         # TODO(simon): enable cross language user config
-        return DeploymentVersionProto(code_version=self.code_version, user_config=b"")
+        return DeploymentVersionProto(
+            code_version=self.code_version,
+            deployment_config=self.deployment_config.to_proto(),
+            replica_config=self.replica_config.to_proto(),
+        )
+
+    def _get_serialized_deployment_config(
+        self, deployment_config: DeploymentConfig
+    ) -> bytes:
+        """Returns a serialized dictionary containing fields of a deployment config that
+        should prompt a deployment version update.
+        """
+        return str.encode(
+            json.dumps(
+                {
+                    "max_concurrent_queries": deployment_config.max_concurrent_queries,
+                    "graceful_shutdown_timeout_s": (
+                        deployment_config.graceful_shutdown_timeout_s
+                    ),
+                    "graceful_shutdown_wait_loop_s": (
+                        deployment_config.graceful_shutdown_wait_loop_s
+                    ),
+                    "health_check_period_s": deployment_config.health_check_period_s,
+                    "health_check_timeout_s": deployment_config.health_check_timeout_s,
```

Comment:
Are these the deployment config options that, if changed, we want to redeploy the replicas?

Reply:
It would be great if we could encode the options that require lightweight/heavyweight updates directly in the code somehow. e.g., an annotation on the `deployment_config` struct definition. can you think of a way to do that?

Reply:
but to answer your question directly, if we are broadcasting these updates to the replicas using the existing logic for `user_config` (as I saw above), then none of these should require restarting the replicas.

Reply:
> It would be great if we could encode the options that require lightweight/heavyweight updates directly in the code somehow. e.g., an annotation on the `deployment_config` struct definition. can you think of a way to do that?

@edoakes I've added it as an extra arg in each pydantic field